modules/playlists/src/main/java/org/opencastproject/playlists/PlaylistRestService.java

```diff
@@ -0,0 +1,772 @@
+      return Response.ok().entity(new JaxbPlaylist(playlist)).build();
+    } catch (Exception e) {
+      return Response.serverError().build();
+    }
+  }
+
+  @POST
+  @Produces(MediaType.APPLICATION_XML)
+  @Path("{id}/entries.xml")
+  @RestQuery(
+      name = "updateEntries",
+      description = "Updates the entries of a playlist",
+      returnDescription = "The updated playlist.",
+      pathParameters = {
+          @RestParameter(name = "id", isRequired = true, description = "Playlist identifier", type = STRING),
+      },
+      restParameters = {
+          @RestParameter(
+              name = "playlistEntries",
+              isRequired = false,
+              description = "Playlist entries in XML format",
+              type = TEXT,
+              jaxbClass = JaxbPlaylistEntry[].class,
+              defaultValue = SAMPLE_PLAYLIST_ENTRIES_XML
+          )
+      },
+      responses = {
+          @RestResponse(responseCode = SC_OK, description = "Playlist updated."),
+          @RestResponse(responseCode = SC_UNAUTHORIZED, description = "Not authorized to perform this action")
+      })
+  public Response updateEntriesAsXml(
+      @PathParam("id") String playlistId,
+      @FormParam("playlistEntries") String entriesText)
+          throws UnauthorizedException {
+    try {
+      // Map JSON to JPA
+      List<PlaylistEntry> playlistEntries = parseXmlToPlaylistEntries(entriesText);
+
+      // Persist
+      Playlist playlist = service.updateEntries(playlistId, playlistEntries);
+      return Response.ok().entity(new JaxbPlaylist(playlist)).build();
+    } catch (Exception e) {
+      return Response.serverError().build();
+    }
+  }
+
+  @POST
+  @Produces(MediaType.APPLICATION_JSON)
+  @Path("{id}/entries/new")
+  @RestQuery(
+      name = "addEntry",
+      description = "Add entry to playlist.",
+      returnDescription = "The playlist with the new entry.",
+      pathParameters = {
+          @RestParameter(name = "id", isRequired = true, description = "Playlist identifier", type = STRING),
+      },
+      restParameters = {
+          @RestParameter(
+              name = "contentId",
+              isRequired = false,
+              description = "Content identifier",
+              type = STRING
+          ),
+          @RestParameter(
+              name = "type",
+              isRequired = false,
+              description = "Entry type. Enum. Valid values are EVENT,"
+              + " INACCESSIBLE.",
+              type = STRING
+          ),
+      },
+      responses = {
+          @RestResponse(responseCode = SC_OK, description = "Playlist updated."),
+          @RestResponse(responseCode = SC_NOT_FOUND, description = "No playlist with that identifier exists."),
+          @RestResponse(responseCode = SC_UNAUTHORIZED, description = "Not authorized to perform this action")
+      })
+  public Response addEntry(
+      @PathParam("playlistId") String playlistId,
```

Comment:
The path annotation uses `id`.
```suggestion
      @PathParam("id") String playlistId,
```